dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -646,6 +646,70 @@ open class Jpql : JpqlDsl {
         return Expressions.ln(value.toExpression())
     }
 
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any> mod(expr: KProperty1<T, Int>, value: Int): Expression<Int> {
+        return Expressions.mod(Paths.path(expr), Expressions.value(value))
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun mod(value1: Expressionable<Int>, value2: Int): Expression<Int> {
+        return Expressions.mod(value1.toExpression(), Expressions.value(value2))
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any> mod(expr: KProperty1<T, Int>, value: Expression<Int>): Expression<Int> {
+        return Expressions.mod(Paths.path(expr), value.toExpression())
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun mod(value1: Expressionable<Int>, value2: Expressionable<Int>): Expression<Int> {
+        return Expressions.mod(value1.toExpression(), value2.toExpression())
+    }
+
+    /**
+     * Create expression that calculates the powering of a numeric [base] to a specified [exponent].
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any, V : Number> power(base: KProperty1<T, @Exact V>, exponent: Int): Expression<V> {
+        return Expressions.power(Paths.path(base), Expressions.value(exponent))
+    }
+
+    /**
+     * Create expression that calculates the powering of a numeric [base] to a specified [exponent].
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Number> power(base: Expressionable<T>, exponent: Int): Expression<T> {
+        return Expressions.power(base.toExpression(), Expressions.value(exponent))
+    }
+
+    /**
+     * Create expression that calculates the powering of a numeric [base] to a specified [exponent].
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any, V : Number> power(base: KProperty1<T, @Exact V>, exponent: Expression<Int>): Expression<V> {
```

Comment:
When used as a parameter to a dsl method, it is more appropriate to write it with the Expressionable<> type instead of the Expression<> type

e.g.
```suggestion
    fun <T : Any, V : Number> power(base: KProperty1<T, @Exact V>, exponent: Expressionable<Int>): Expression<V> {
```

Reply:
Is it to accept a wider range of material types?? reflected ! :)

Reply:
Thanks for checking it out. 
I'm actually pretty much done with it, but I've had a little bit of a break, so I haven't had a chance to do a final review and PR.
I'll probably PR it really soon.
Thanks.